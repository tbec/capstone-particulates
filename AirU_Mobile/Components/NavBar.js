import React, {Component} from 'react';
import {Text, View, TouchableOpacity, Image} from 'react-native';
import {styles} from '../StyleSheets/Styles'
import {Navigation} from './Navigation'

export default class NavBar extends Component<Props>
{
    render(){
        return(
            <View style={styles.navBar}>
                <Text>NavBar</Text>
                {/* settings */}
                <TouchableOpacity onPress={() => Navigation.push('Settings')}>
                    <Image source={require("../Resources/Sensor.png")}/>
                </TouchableOpacity>
            </View>
        )
    };
}