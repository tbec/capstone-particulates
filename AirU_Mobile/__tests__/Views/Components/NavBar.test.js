import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import NavBar from '../../../Components/NavBar'

let findById = function(tree, testID) {
    if(tree.props && tree.props.testID === testID) {
        return tree
    }
    if(tree.children && tree.children.length > 0)
    {
        let childs = tree.children
        for(let i = 0; i < childs.length; i++)
        {
            let item = findById(childs[i], testID)
            if(typeof(item) !== 'undefined') {
                return item
            }
        }
    }
}

test('Snapshot NavBar Setup', () => {
    const snap = renderer.create(<NavBar navigation={undefined}/>).toJSON();
    expect(snap).toMatchSnapshot();
})

test('NavBar with nextButton', () => {
    const nav = renderer.create(<NavBar previous='ConnectionSetup'/>).toJSON();
    button = findById(nav, 'prevButton')
    expect(button).toBeDefined();
    button = findById(nav, 'nextButton')
    expect(button).toBeUndefined();
})

test('NavBar with nextButton', () => {
    const nav = renderer.create(<NavBar next='ConnectionSetup'/>).toJSON();
    button = findById(nav, 'nextButton')
    expect(button).toBeDefined();
    button = findById(nav, 'prevButton')
    expect(button).toBeUndefined();
})

test('NavBar with both', () => {
    const nav = renderer.create(<NavBar next='ConnectionSetup' previous='ReviewFirst'/>).toJSON();
    button = findById(nav, 'nextButton')
    expect(button).toBeDefined();
    button = findById(nav, 'prevButton')
    expect(button).toBeDefined();
})