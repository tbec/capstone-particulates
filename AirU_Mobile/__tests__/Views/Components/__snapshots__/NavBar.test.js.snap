// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot NavBar Setup 1`] = `
<View
  style={
    Object {
      "alignContent": "space-between",
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "flex-end",
      "width": "100%",
    }
  }
/>
`;
